src/pages/documentation/resources/upgrade-platform-sdks/tabs/analytics.md

```diff
@@ -0,0 +1,245 @@
+// AppDelegate.h
+@import AEPCore;
+@import AEPIdentity;
+@import AEPAnalytics;
+
+// AppDelegate.m
+- (BOOL) application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+  [AEPMobileCore registerExtensions:@[AEPMobileAnalytics.class, AEPMobileIdentity.class] completion:^{
+      // Use the environment file id assigned to this application in Data Collection UI
+      [AEPMobileCore configureWithAppId: @"your-environment-file-id"];
+  }];
+  return YES;
+}
+```
+
+<Variant platform="android" task="api-changes-v4" repeat="4"/>
+
+#### Java
+
+The Adobe Mobile Library (v4) syntax and usage examples for these API are:
+
+```java
+// syntax
+public static void trackState(final String state, final Map<String, Object> contextData)
+
+// usage
+Analytics.trackState("MainPage", new HashMap<String, Object>() {{
+  put("firstVisit", true);
+}});
+```
+
+```java
+// syntax
+public static void trackAction(final String action, final Map<String, Object> contextData)
+
+// usage
+Analytics.trackAction("linkClicked", new HashMap<String, Object>() {{
+  put("url", "https://www.adobe.com");
+}});
+```
+
+<Variant platform="ios" task="api-changes-v4" repeat="4"/>
+
+#### Objective-C
+
+The Adobe Mobile Library (v4) syntax and usage examples for these API are:
+
+```objectivec
+// syntax
++ (void) trackState:(NSString *)state data:(NSDictionary *)data;
+
+// usage
+[ADBMobile trackState:@"MainPage" data:@{@"firstVisit":@true}];
+```
+
+```objectivec
+// syntax
++ (void) trackAction:(NSString *)action data:(NSDictionary *)data;
+
+// usage
+[ADBMobile trackAction:@"linkClicked" data:@{@"url":@"https://www.adobe.com"}];
+```
+
+<Variant platform="android" task="api-changes-aep" repeat="4"/>
+
+The Mobile SDKs have moved the `trackAction` and `trackState` APIs to the MobileCore extension. In addition, the context data Map has been changed from `<String, Object>` to `<String, String>`. The syntax is:
+
+#### Java
+
+```java
+// syntax
+public static void trackState(final String state, final Map<String, String> contextData)
+
+// usage
+MobileCore.trackState("MainPage", new HashMap<String, String>() {{
+  put("firstVisit", "true");
+}});
+```
+
+```java
+// syntax
+public static void trackAction(final String action, final Map<String, String> contextData)
+
+// usage
+MobileCore.trackAction("linkClicked", new HashMap<String, String>() {{
+  put("url", "https://www.adobe.com");
+}});
+```
+
+<Variant platform="ios" task="api-changes-aep" repeat="8"/>
+
+The Mobile SDKs have moved the `trackAction` and `trackState` APIs to the MobileCore extension. In addition, the context data dictionary has been changed from `<NSString, NSObject>` to `<NSString, NSString>`. The syntax is:
+
+```objectivec
++ (void) trackAction: (nullable NSString*) action data: (nullable NSDictionary<NSString*, NSString*>*) data;
+```
+
+```objectivec
++ (void) trackState: (nullable NSString*) action data: (nullable NSDictionary<NSString*, NSString*>*) data;
```

Comment:
Let us be consistent and provide the Swift syntax/example and objective C syntax/ example (and in that order).

Reply:
added the syntax from mobilecore api-reference